file/builder.go

```diff
@@ -890,6 +962,27 @@ func (b *stateBuilder) plugins() {
 	}
 }
 
+func (b *stateBuilder) validatePlugin(p FPlugin) error {
+	if (b.isConsumerGroupScopedPluginSupported && !b.isKonnect) && *p.Name == ratelimitingAdvancedPluginName {
+		// check if deprecated consumer-groups configuration is present in the config
+		var consumerGroupsFound bool
+		if groups, ok := p.Config["consumer_groups"]; ok {
+			// if groups is an array of length > 0, then consumer_groups is set
+			if groupsArray, ok := groups.([]interface{}); ok && len(groupsArray) > 0 {
+				consumerGroupsFound = true
+			}
+		}
+		_, enforceConsumerGroupsFound := p.Config["enforce_consumer_groups"]
+		if consumerGroupsFound || enforceConsumerGroupsFound {
+			return errors.New("a rate-limiting-advanced plugin with config.consumer_groups\n" +
+				"and/or config.enforce_consumer_groups was found. Please use Consumer Groups scoped\n" +
+				"Plugins when running against Kong Enterprise 3.4.0 and above.\n\n" +
+				"Check DOC_LINK for more information")
```

Comment:
Perhaps we could put this error in a variable which could then be compared against at the call site?